ndc-client/src/apis/default_api.rs

```diff
@@ -271,18 +246,26 @@ pub async fn schema_get(
             if !response_status.is_client_error() && !response_status.is_server_error() {
                 serde_json::from_value(response_content).map_err(Error::from)
             } else {
-                let error_response: crate::models::ErrorResponse =
-                    serde_json::from_value(response_content)?;
-                let connector_error = super::ConnectorError {
-                    status: response_status,
-                    error_response,
-                };
-                Err(Error::ConnectorError(connector_error))
+                handle_error(response_status, response_content)
             }
         })
         .await
 }
 
+fn handle_error<T>(
```

Comment:
We don't need the generic `T` here, since we always return `Err` anyway, so we can just return `Error` and move the `Err` wrapper to the call-site.

Reply:
Done.